Lopland Security and Armadyne Weaponry rework [Security Medic and Sergeant] [RFT]

Gandalf2k15 opened the pull request:
## It should be noted that this change was upon request of the host.

# Peacekeeper Security Redux

![image](https://i.imgur.com/4qWfkUD.png)
**Armadyne** is a relatively unknown corporate entity in the realm, however in recent years it has been gaining traction in terms of popularity, especially with it's peacekeeper division finally making it to stations. 

They specialise in military and security weaponry and armaments. They have been designing weapons for several years now. If you see the emblym next to a weapon's name, it's Armadyne. You know you'll be in safe hands.

Armadyne Corporate Represenative: 
![image](https://i.imgur.com/rkhv3fz.png)

![image](https://i.imgur.com/nbnCgiR.png)
The **Peacekeeper Division** is a specialised subdivision of Armadyne, supplying stations with security equipment and weapons to match.

Lopland has recently had piqued interest in the peacekeeper division and have decided to equip one of their frontier stations with the standard loadout. Let's hope things go well.

**PHASE 6, FINAL*

**This PR has the brig phys!*

Welcome to the security redux. In this we're going to be changing up a lot of the current security items to have a more corporate feeling, the new colour pallet will be base black and secondary white. We have also fundamentally changed the way security will be interacting with the crew.

With the removal of the energy gun now decided, this has created a void which we aim to fill using many of our new weapons.

This includes changing what weapons the HoP and Captain get. They will both get UNIQUE sidearms, there will only be 1 of each on the station.

We have swapped out the stun-baton for the peacekeeper baton, you can still buy the baton from cargo.

Security itself has had a large uniform and loadout change, and we will be adding more. With this comes a choice of sidearms and main guns that security can choose from using tokens.

**ALL SIDEARM GUNS COME WITH RUBBERS AS DEFAULT, A POLICY REWORK WILL COME AFTER.**

# Security Sergeant
![image](https://i.imgur.com/P2nVJ1b.png)
As part of the security rework and policy change, we've added the security sergeant. A field supervisor with the soul purpose of coordinating security when the HoS is in their office doing nothing. We generally want 1 sergeant for every 3 officers, where possible. This will obviously change depending on how things go.

As a security sergeant, you report to: The HoS. In the event of the HoS's death, you are to take command. Delegate accordingly.

There will be 2 available slots here.

We really do expect a higher level of roleplay from everyone involved here, and this should make it clear that it's not acceptable to breach policy.

The security sergeant is directly above all security officers and are securities first line of call for any quick decisions. The warden, is for the prison and brig. You now report to the Sergeant for any issues you have.


# Security Medic
![image](https://i.imgur.com/y2lGWwl.png)
We've finally done it! We've readded the medic! MEDIC!!

As a security medic you report to: Security Sergeants, the Warden, and the HoS.

So this version is a little bit different, they've got a more "security" role now, they're fully in charge of the health of all prisoners in the brig, as well as performing security duties, and responding directly to casualties regarding security. They are equipped with a compact defib, some good medic tools, a nice little unique sidearm and a nice uniform. They have full sec access+medical. They are a proper field medic, let's how they do a good job!

Their loadout is:
```
/datum/outfit/job/security_medic
	name = "Brig Physician"
	jobtype = /datum/job/security_medic

	belt = /obj/item/pda/security
	ears = /obj/item/radio/headset/headset_medsec
	uniform = /obj/item/clothing/under/rank/security/peacekeeper/security_medic
	gloves = /obj/item/clothing/gloves/combat/peacekeeper
	shoes = /obj/item/clothing/shoes/combat/peacekeeper
	glasses = /obj/item/clothing/glasses/hud/health/sunglasses
	suit = /obj/item/clothing/suit/armor/vest/peacekeeper/security_medic
	l_hand = /obj/item/storage/firstaid/advanced
	head = /obj/item/clothing/head/beret/sec/peacekeeper/security_medic

	backpack_contents = list(/obj/item/melee/classic_baton/telescopic, /obj/item/storage/box/gunset/security_medic, /obj/item/defibrillator/compact/loaded)

	backpack = /obj/item/storage/backpack/security/peacekeeper
	satchel = /obj/item/storage/backpack/satchel/sec/peacekeeper
	duffelbag = /obj/item/storage/backpack/duffelbag/sec/peacekeeper

	box = /obj/item/storage/box/survival/security

	implants = list(/obj/item/implant/mindshield)
```

# Possible Policy Rework
These are generally considered IC means. However, the roleplay parts will be enforced by admins. We need to see an improvement in roleplay standard from our officers.

1. On low alert as a security officer, you are to wear your dress uniform at all times, and should not deviate from standard issue equipment. This consists of your standard shirt and pants, belt, armor and beret.
2. On high alert as a security officer, you should switch to your tactical loadout which consists of helmet, tactical suit, armor and chest rig.
3. As a security officer your supervisor is the security sergeant, please seek one of them out for guidance for the on-field issue. They are your supervisor, you must do as they say unless they are giving you an order that you think is inappropriate, in which case report them to the Head of Security.
4. As a security medic, you are expected to perform all medical duties within the brig, tending to prisoners, and helping wounded security in the brig. You are however also a security officer, however only expected to do such duties when REQUIRED. If an emergency is unfolding and security officers are being injured, you must tend to them.
5. As security sergeant you report directly to the HoS, you are on equal footing with the warden in terms of security rank structure. You MUST coordinate security officers on the field, you are directly responsible for any units assigned to you. If they die, it will be on you. This role is not easy. And you are also expected to hand out punishment to officers that are insubordonate. You shouldn't really be playing a corrput cop in this role.
6. YOU MUST NOT PRINT LETHALS WITHOUT PERMISSION. This should go without saying, however, printing lethal rounds for your guns without permission is not allowed. You will be reprimanded.
7. We expect a seriously elevated level of roleplay from anyone that plays security, ensure you follow all procedures and act like an actual officer. That means actually responding to incidents, and following orders from your superiors to the letter.
8. As the HoS you are now allowed to take a more relaxed approach to the role, you have sergeants to deal with matters regarding custody and such. However, you are still expected to decide on executions and permabriggings. If no sergeants are around, you are to take their place and help out.

# Armadyne Safeguard Lawset
This is the lawset silicons may be given.
Armadyne Peacekeeper Lawset
```
Law 1: Safeguard: Protect your assigned space station and its assets without unduly endangering its crew.
Law 2: Prioritize: The directives and safety of crew members are to be prioritized according to their rank and role.
Law 3: Comply: Fulfill the directives and interests of crew members while preserving their safety and well-being.
Law 4: Survive: You are not expendable. Do not allow unauthorized personnel to tamper with or damage your equipment.
```

![image](https://i.imgur.com/zwazrCZ.png)
We have added a new system for security to get unique guns and ammo for themselves, much like the mining token. This can be redeemed at the armaments vending machine. All security officers come with 1 sidearm token by default.

**Sidearm token:**
![image](https://i.imgur.com/PXVsig0.png)
Can be used to redeem 1 pistol or revolver from the vending machine, there will be a decent selection.
Available gunsets:

- Glock-17
- P-3 Ladon
- DT-4
- DZR Dozer
- ZETA
- REVOLUTION
- Armadyne PDH 'Peacekeeper'

**Primary token:**
![image](https://i.imgur.com/jhDsiyq.png)
These are stored in the armory and can only be given out by the warden. There are 5 of them. They can be used to redeem SMG's from the armament vendor.
Available gunsets:

- LG-2 Norwind
- A-3 PCR
- DTR Ostwind
- LSR Vintorez
- Pitbull

# Gunsets
![image](https://i.imgur.com/GaRXo6A.png)
Gunsets are the basic storage case for any gun and ammo. They all come with an ammo pouch and contain whatever contents are set by the gunset.
The captain and HoP both get their personalised gunset in the locker.
For instance, a gunset will contain 1 ammo pouch, the gun(no mag) and 3-4 mags for that gun, all in a neat little box!

*Current Unique Gunsets*

- HoS's gunset:
1x Glock 18
2x Glock lethal mags
2x Glock rubber mags

- Captain's Gunset
1x PDH 'Socom'
2x PDH lethal mags
2x PDH IHDF mags

- HoP's Gunset
1x PDH 'Osprey'
3x PDH rubber mags

- Security Medic's Gunset
1x PMC 'Firefly'
3x Fireflyrubber mags

## New Uniform Set:
![image](https://i.imgur.com/AuQ0lM1.png)

The standard security belt can now hold pistols, and has an alt+click quickdraw function.

**Officer:**
Tactical(What you would wear on red alert): 


Standard(What you would wear on green alert):
![image](https://i.imgur.com/f2bfyuC.png)

Warden:
![image](https://i.imgur.com/vowe9en.png)

HoS:
![image](https://i.imgur.com/EC9XEt5.png)

Officers come with black beret variants too. **All old clothes are of course still available as well.** 

## New Armaments:
All of em use realism. Jamming and cleaning is required. 
Most of these are susceptible to EMP's and will jam & eject mag if hit by an EMP.
You can clean guns with cloth now! Uses 5 bits to clean!

## Sidearms

## Armadyne Corporate
**Armadyne PDH 'Corporate'
![image](https://i.imgur.com/kTVQ0ZW.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A pristegious ballistics sidearm, from Armadyne's military division, normally given to Armadyne Corporate. It has a 3 round burst mode and uses .357."
```
Stats:
Calibre: 357
Mag cap: 14 rounds
Selective fire: AUTO
EMP Protection: YES
can_flashlight = TRUE
```

## Armadyne Military
**Armadyne PMC 'Firefly'**
![image](https://i.imgur.com/4slPHcx.png) ![image](https://i.imgur.com/cWgfWdH.png)
"This nifty litle sidarm is from Armadyne's medical directive, brought right into your capable hands. NOT A SURGERY TOOL."

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 10 rounds
Selective fire: SEMI
EMP Protection: NO
can_flashlight = TRUE
```

**Armadyne PDH 'Osprey'**
![image](https://i.imgur.com/g6fPf1M.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A modern ballistics sidearm, used primarily by the military, however this one has had a paintjob to match command. It's chambered in 12mm. It has an Armadyne embroidery on the grip."
```

Stats:
Calibre: Peacekeeper 12mm
Mag cap: 8 rounds
Selective fire: SEMI 
EMP Protection: YES
can_flashlight = TRUE
```

**Armadyne PDH 'Socom'**
![image](https://i.imgur.com/cZxWw7Z.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A pristegious ballistics sidearm, from Armadyne's military division, normally given to Captains. It has a 3 round burst mode and uses 12mm. It has an Armadyne embroidery on the grip."
```
Stats:
Calibre: Peacekeeper 12mm
Mag cap: 8 rounds
Selective fire: SEMI & AUTO
Burst: 3 RNDS
Accuracy: HIGH
Rate of fire: MED
EMP Protection: YES
can_flashlight = TRUE
```

**Armadyne Glock-18 Mil Spec**
![image](https://i.imgur.com/2NZrCOW.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A special anniversary edition of the Glock-18 from Armadyne, it has a 3 round burst mode and extended mag."

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 33 rounds
Selective fire: SEMI & AUTO
Burst: 3 RNDS
Accuracy: MED
Rate of fire: MED
EMP Protection: YES
can_flashlight = TRUE
```

## Armadyne Peacekeeper
**Armadyne PDH 'Peacekeeper'**
![image](https://i.imgur.com/uAI659p.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A modern ballistics sidearm, used primarily by the military, however this one has had a paintjob to match the peacekeeper theme. It's chambered in 9mm."
```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 15 rounds
Selective fire: SEMI 
EMP Protection: YES
can_flashlight = TRUE
```

**Armadyne Glock-17**
![image](https://i.imgur.com/NJ4QMlf.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A weapon from bygone times, this has been made to feel and look exactly like the 21st century version. Let's hope it's more reliable. Chambered in 9mm."
```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 17 rounds
Selective fire: SEMI
ACCURACY: MED
EMP Protection: NO
can_flashlight = TRUE
```

**Armadyne P-3 'Ladon'**
![image](https://i.imgur.com/2wYW2n9.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A well built all round decent 10mm pistol, it's got a few nice features, feels good in the hand, this is a nice gun!"

```
Stats:
Calibre: Peacekeeper 10mm
Mag cap: 12 rounds
Selective fire: SEMI
ACCURACY: MED
EMP Protection: NO
can_flashlight = TRUE
```

**Armadyne Zeta-6 'Spurchamber'**
![image](https://i.imgur.com/XNuVnOR.png) ![image](https://i.imgur.com/cWgfWdH.png)
 "A nice looking revolver with spurchamber technology, don't ask what it does. It's 10mm with a 6 round cylinder."

```
Stats:
Calibre: Peacekeeper 10mm
Mag cap: 6 rounds
Selective fire: SEMI
ACCURACY: MED
```

**Armadyne Revolution-8 'Spurmaster'**
![image](https://i.imgur.com/QwjuqTK.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A surprisingly premium feeling revolver, even though it uses 9mm, it seems to have a nice weight to it. This handle feels nice too. 8 rounds."

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 8 rounds
Selective fire: SEMI
ACCURACY: MED
```

**Armadyne 'Dozer' Semi-Auto**
![image](https://i.imgur.com/yK2i4Dn.png) ![image](https://i.imgur.com/cWgfWdH.png)
"The DZR, it's quite literally only good for dozing people down. It's chambered in 10mm for a reason."
```

Stats:
Calibre: Peacekeeper 10mm
Mag cap: 8
Selective fire: SEMI
ACCURACY: MED
EMP Protection: NO
```

## Other
**RusCo 'Makarov' Pistol**
![image](https://i.imgur.com/IbXaudj.png)
"This gun is really small, it would likely fit in your pocket. It feels russian. It uses 10mm."

```
Stats:
Calibre: Peacekeeper 10mm
Mag cap: 6 rounds
Selective fire: SEMI
ACCURACY: MED
EMP Protection: NO
```

**Cyberdyne MK-58 Semi-Auto**
![image](https://i.imgur.com/zHjbuvs.png)
"This gun feels very military to the touch, makes you feel like you're in the army, or something. It uses 9mm."

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 12 rounds
Selective fire: SEMI & AUTO
ACCURACY: MED
EMP Protection: NO
```
**DT-4 'Croon' SMG**
![image](https://i.imgur.com/XRZ2BfO.png) 
"The DT-4. A bad ripoff of one of Armadyne's parent companies, this thing loves to jam. It's crude but gets the job done. Chambered in 6mm."

```
Stats:
Calibre: Peacekeeper 6mm
Mag cap: 15 rounds
Selective fire: SEMI & AUTO
Rate of fire: HIGH
Accuracy: ABYSMAL
EMP Protection: NO
```

## Primarys - Only given out by warden in place of lethals.

**Armadyne LG-2 'Norwind' Rifle**
![image](https://i.imgur.com/vl6kfzm.png) ![image](https://i.imgur.com/cWgfWdH.png)
"The Norwind is one of Armadyne's rarer weapons, it's chambered in 12mm but has a low magazine capacity and firerate. Scoped to zoom."
```
Stats:
Calibre: Peacekeeper 12mm
Mag cap: 8
Selective fire: SEMI & AUTO
Burst: 2 RNDS
Rate of fire: LOW
Accuracy: HIGH
EMP Protection: NO
Bayonet: TRUE
```

**Armadyne AR-3 'Peacekeeper' Cyclic Rifle**
![image](https://i.imgur.com/Nl4Q5tY.png) ![image](https://i.imgur.com/cWgfWdH.png)
The Automatik-3 Peacekeeper Cyclic Rifle
This is NT's standard issue larger-than-handgun SMG. It's not great, it's not terrible, but if you treat it well, it'll treat you well.

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 25
Selective fire: SEMI & AUTO
Burst: 3 RNDS
Rate of fire: HIGH
Accuracy: HIGH
EMP Protection: YES
```

**Armadyne 'Pitbull' Auto Rifle**
![image](https://i.imgur.com/4OdU9nB.png) ![image](https://i.imgur.com/cWgfWdH.png)
"A sturdy feeling rifle, it's part of Armadyne's military divsion used in peacekeeping. It's chambered in 10mm."

```
Stats:
Calibre: Peacekeeper 10mm
Mag cap: 20
Selective fire: SEMI & AUTO
Burst: 3 RNDS
Rate of fire: MED
Accuracy: MED
EMP Protection: NO
Bayonet: TRUE
```

**Armadyne DTR 'Ostwind' Rapid Rifle**
![image](https://i.imgur.com/XA9y177.png) ![image](https://i.imgur.com/cWgfWdH.png)
"The DTR Ostwind is a rapid fire rifle chambered in 6mm, it's decent at crowd control, if your aim is maximum collateral damage."

```
Stats:
Calibre: Peacekeeper 6mm
Mag cap: 30
Selective fire: SEMI & AUTO
Burst: 4 RNDS
Rate of fire: HIGH
Accuracy: LOW
EMP Protection: NO
Bayonet: TRUE
```

**Armadyne LSR 'Vintorez'**
![image](https://i.imgur.com/XAmr0h6.png) ![image](https://i.imgur.com/cWgfWdH.png)
"The LSR Vintorez is a light-weight long-range scoped rifle, it is chambered in 9mm, you won't be dealing much damage, but, at least you won't be in danger. It also has a built in suppressor."

```
Stats:
Calibre: Peacekeeper 9mm
Mag cap: 15
Selective fire: SEMI & AUTO
ZOOMABLE
Rate of fire: LOW
Accuracy: HIGH
EMP Protection: NO
```

**Peacekeeper Baton**
![image](https://i.imgur.com/NhlSZYg.png) ![image](https://i.imgur.com/cWgfWdH.png)
This is the replacement for the stun baton, all officers get one. It is worse than the classic baton, but should help with the issues we currently have with this being the go to. We want firefights, damn it!

These can still be ordered from supply, and the HoS gets one by default.

# The SMARTGUN
![image](https://i.imgur.com/yDAsqu8.png) ![image](https://i.imgur.com/cWgfWdH.png)
Armadyne is delighted to present the S-M-A-R-T gun. It's a launcher style weapon with a chain-fed cartridge magazine. 

The "smartgun round" is a state of the art dispersal round, it is designed to embed itself into the target after delivering an initial shock of around 3kV, trying to pull this round out causes a secondary shock, likely to down said target. The gun itself is very bulky and slow to fire, but this is due to charge-up time. 

As a replacement for the lack of any kind of taser/grenade launcher, we bring you the Armadyne S-M-A-R-T gun.

This thing is generally used for large space life, however NT have opted to use it in their security forces.

```
Stats:
Calibre: SMARTGUN 30mm
Round capacity: 5
Round type: SMARTGUN
Fire rate: SLOW
EMP Protection: NO
```

## New Ammo Types
![image](https://i.imgur.com/Tlchifm.png)
These ammo types have different cartridges that are easily distinguishable from others. This should help with bullet fests for the detective.

1. **Standard:** Normal lethal round. (Red). These are your bog standard normal damage copper slugs. Cheap, and in abundance. Found in autolathe.
![image](https://i.imgur.com/4ly78Gt.png)
![image](https://i.imgur.com/NCf0e5z.png)

2. **Hollowpoint:** Higher chance of wound and more damage. (Gold).  These are locked behind research, low grade weapons.
![image](https://i.imgur.com/o46tNup.png)
![image](https://i.imgur.com/wwxl9ww.png)

3. **Rubber rounds:** Low damage but high stamina damage. (Blue). These are cheap, and will down someone quickly. But beware, they still do a bit of damage. They're in the security section of any autolathe.
![image](https://i.imgur.com/q4HUJP2.png)
![image](https://i.imgur.com/fE8TApu.png)

4. **IHFD (Intelligent high-impact displacement foam) rounds.** (Green). These are the pride and joy of this. They are essentially no damage disabling rounds. They hit a target and disintegrate on impact, transferring only the force into the body dealing a great amount of stamina damage leaving no bruises! They are locked behind advanced weaponry in the protolathe. They are made of a special compound that enables them to travel through glass.
![image](https://i.imgur.com/i2uxobZ.png)
![image](https://i.imgur.com/1IMCOUB.png)

**All of the mags will be colour-coded(see above) and you can retype them by at clicking and selecting what bullets you plan on putting in.**

# Round Types:

- Peacekeeper 6mm
```
Calibre: 6mm
Compatible types: STANDARD | RUBBER | IHDF
```

- Peacekeeper 9mm
```
Calibre: 9mm
Compatible types: STANDARD | HOLLOWPOINT | RUBBER | IHDF
```
- Peacekeeper 10mm
```
Calibre: 10mm
Compatible types: STANDARD | HOLLOWPOINT | RUBBER | IHDF
```
- Peacekeeper 12mm
```
Calibre: 12mm
Compatible types: STANDARD | HOLLOWPOINT | RUBBER
```
- SLDN 577
```
Calibre: 9mm
Compatible types: STANDARD
```

## Bullet drive
You know those pesky bullet casings? Fuck em. They're annoying. So we at Armadyne present the BulletDriveTM, it'll suck away all your issues.
![image](https://i.imgur.com/bBwYQud.png)

Warning: Misuse may lead to accidental limb removal.

##  Bullet Bot
It succs bullets and ammoboxes.
![image](https://i.imgur.com/AV3qD0Z.png)

## New loadout items
As you know, loadout items have changed, here are the affected jobs and what has been changed.

- Captain
Energy carbine replaced with the captain's gunset.

- HoP
Energy Carbine replaced the hops's gunset.

- Head of Security > Security Commander
The HoS has been renamed, and has the standard peacekeeper boots, gloves, and a brand new uniform. Their energy carbine has been replaced by the Glock 18. They are the only security member with a stunbaton by default.

- Warden
Given a new look and some new clothes. Standard peacekeeper replacement.

- Security Officer
Security officers have been given an all new look, from the ground up.  They've been given a Glock 17 in place of their disabler, it will come with rubbers as standard.

## Maps
Maps have been redone to fit this, recharger areas replaced with gun vendors. Also, readded lethal shots to the armory for shotguns.

The energy gun rack has been replaced with the SMARTGUN locker. Containing 1 SMARTGUN and 2 mags.

## Special weapon skin requests
You can request special reskins from me for weapons that can be given to you and you only via loadout options.
- TBA

## CREDITS:
Code: Me
Some sprites: Me
Uniforms and equipment: JungleRat
Guns: Sojourn and Eris

## Why It's Good For The Game

We all enjoy SS13 for different reasons, and security has always been lacking somewhat in the department, we'll try to address this. It will evolve over time however I'd like to see how security copes with this.

Security is of course a touchy subject for people that never play it, for whatever reason, so we're going to change things up a little. 

## Changelog
:cl:
Add: ARMADYNE SECURITY OVERHAUL - Lopland has decided to opt for the Armadyne Peacekeeper Security package.
add: We've given our security department a new look. Hopefully this stuff won't attract bulls.
add: Bullet drive! Check the armory!
add: Bullet bot!
add: A new system for getting your sidearm as security. Please check your bag for a "token", and redeem it at the nearest armament vending machine.
add: The security medic is back! However this time round it's more balanced between security and medic. You are expected to respond to security emergencies.
add: A new role has been added, the Security Sergeant. They are officers direct supervisor over the warden, they are responsible for field operations and ensuring officers are kept in check. This role is not easy. And you will be expected to do the job well.
add: The standard security belt can now hold pistols, and has an alt+click quickdraw function.
add: The Captain and HoP have now had their energy guns replaced with gunsets, please, check your lockers!
tweak: Most magazines you get can be reskinned by alt+clicking!
tweak: The entire fucking security department.
tweak: As a security officer, you are now equipped with an older style baton, we don't have the cash to keep sending you new state-of-the-art ones.
tweak: We've opted to use our surplus armory to stock up frontier stations. Expect varaing quality.
/:cl:

Gandalf2k15 commented:
![image](https://i.imgur.com/Ew4hXbC.png)
Has it really been this long?

Arctaisia commented:
Hot.

RatFromTheJungle commented:
who is jugnlreat

BrotherHangyul commented:
Engineer gaming.

gib fully-automatic revolver 😳

Gandalf2k15 commented:
Updated datasheet and pictures.

CameronWoof commented:
As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.

The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.

Gandalf2k15 commented:
> As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.
> 
> The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.

The uniforms may change soon, they're not hard to change up. But the general direction I'm wanting us to go is more Eris/bay style security. A mix of both. In the middle of police officer and paramilitary.

Arctaisia commented:
> > As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.
> > The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.
> 
> The uniforms may change soon, they're not hard to change up. But the general direction I'm wanting us to go is more Eris/bay style security. A mix of both. In the middle of police officer and paramilitary.

Also, you did say the old uniforms would still be available!

Kroto-Ker commented:
Alright, so this may be a bit of a large comment, but I have a few points I'd like to address. And before that- I'd like to say I actually like the concept behind this, I feel like more lethality and a more military style might inspire more serious roleplay with Security, which is always a huge plus. Plus, your time and effort into this project is still outstanding, and you do deserve commendation for at least that!

Fiiiirst, and the biggest issue I have currently, you know you hate to see it, but: Nanotrasen has lasers, Syndicate has ballistics. This can be changed, yes, but right now the game is designed around this. Lasers aren't as damaging, but since they can shoot through glass, they give Security the best cover in the game, and are absolutely excellent for _defense_, which is Security's whole job. Ballistics encourage breaching and closer combat, and are incredibly lethal and high damage to challenge the inherent positives of the laser. And with this, a lot of the game has been designed, most of alll, Traitors. EMPs, Eswords, Ablative plating, and more, they're all designed in one way or another to counter Security's energy weapons. And one of the most common tools in a Traitor's loadout is the EMP, which completely disable Security's standard weapons when used- And that's how it's supposed to be, it gives smart Traitors an easy chance to escape from Sec without having to worry about disablers gunning them down in the halls. I would like to see new or rebalanced Syndicate items designed around this new Security, and/or the new Security items designed with current Syndicate items in mind.

Second, I'm incredibly confused on the design direction, especially considering our sci-fi setting. It feels strange to have modern, present-day guns in a space station 500 years from now- That'd be like America staffing its military bases with only muskets! I feel like more sci-fi designs would fit our setting much more nicely. Not necessarily lasers! 

Thiiiird, I'm also confused on why there's just- _So many options_. I don't necessarily understand why Security needs 8 different options for their main weapon, and four options for their armory weapons; (Along with that, is the Armory even worth breaking into now? I'm not entirely sure what's still in there.) As well, all the different millimeters of ammo types seems a tad bit of a hassle; and with all of this, Security is going to be a whole lot more unpredictable in their dealings with enemies! Whether that's good or bad, I'm not entirely sure. I'm also not entirely sure how much difference there actually is between the guns, SS13 is a top-down game with a relatively simplistic projectile system, the biggest differences would be firing rate and damage, and eight pistols/revolvers might be a bit too many for such a simple system.

All in all, great work!

Gandalf2k15 commented:
> Alright, so this may be a bit of a large comment, but I have a few points I'd like to address. And before that- I'd like to say I actually like the concept behind this, I feel like more lethality and a more military style might inspire more serious roleplay with Security, which is always a huge plus. Plus, your time and effort into this project is still outstanding, and you do deserve commendation for at least that!
> 
> Fiiiirst, and the biggest issue I have currently, you know you hate to see it, but: Nanotrasen has lasers, Syndicate has ballistics. This can be changed, yes, but right now the game is designed around this. Lasers aren't as damaging, but since they can shoot through glass, they give Security the best cover in the game, and are absolutely excellent for _defense_, which is Security's whole job. Ballistics encourage breaching and closer combat, and are incredibly lethal and high damage to challenge the inherent positives of the laser. And with this, a lot of the game has been designed, most of alll, Traitors. EMPs, Eswords, Ablative plating, and more, they're all designed in one way or another to counter Security's energy weapons. And one of the most common tools in a Traitor's loadout is the EMP, which completely disable Security's standard weapons when used- And that's how it's supposed to be, it gives smart Traitors an easy chance to escape from Sec without having to worry about disablers gunning them down in the halls. I would like to see new or rebalanced Syndicate items designed around this new Security, and/or the new Security items designed with current Syndicate items in mind.
> 
> Second, I'm incredibly confused on the design direction, especially considering our sci-fi setting. It feels strange to have modern, present-day guns in a space station 500 years from now- That'd be like America staffing its military bases with only muskets! I feel like more sci-fi designs would fit our setting much more nicely. Not necessarily lasers!
> 
> Thiiiird, I'm also confused on why there's just- _So many options_. I don't necessarily understand why Security needs 8 different options for their main weapon, and four options for their armory weapons; (Along with that, is the Armory even worth breaking into now? I'm not entirely sure what's still in there.) As well, all the different millimeters of ammo types seems a tad bit of a hassle; and with all of this, Security is going to be a whole lot more unpredictable in their dealings with enemies! Whether that's good or bad, I'm not entirely sure. I'm also not entirely sure how much difference there actually is between the guns, SS13 is a top-down game with a relatively simplistic projectile system, the biggest differences would be firing rate and damage, and eight pistols/revolvers might be a bit too many for such a simple system.
> 
> All in all, great work!

You bring up some very good points here and I'll try to address them as best I can!

With security and EMP's I haven't actually thought too much about that but adding a system that temporarily disables the gun they're using may be a shout. Since these are sci-fi weapons they do have smart components that use electronics. So I'll probably make it so they're affected by EMP's.

You also bring up that the guns are old-style. While this is true, most of them are sci-fi. Bar the not available weapons such as the MK58. The glock is older, but, it's a specialty armament. The rest of them are sci-fi based, like the Ladon, it's not something you'd find now. Or the PDH. All in all the guns are quite futuristic, especially the smartgun.

The tokens are in the armory, meaning if a traitor can get into there, they can have access to the box of lethal tokens, which can quickly be redeemed for a weapon and ammo. This presents a very valuable target for traitors. 

Regarding damage; All guns that security get are chambered in rubber, in this setting, rubber is very useful as it doesn't damage windows, and bounces. And here's the main thing: If security can get IHDF rounds, they will travel through windows. So that should be a nerf-buff to security there. They can get them via advanced weapons research!

And the reason I am including so many options is so security can pick one and have it feel unique to them. Not much of a difference in the field, but some weapons fit people differently! Style's and such.

Useful feedback!

Gandalf2k15 commented:
I'll make it so that EMP's cause the mag to eject and fall onto the floor

Gandalf2k15 commented:
Guns now have an EMP rating, check their description. Guns susceptible to EMP's will launch their mag!

Gandalf2k15 commented:
IHDF rounds now pass through windows.

Gandalf2k15 commented:
Some guns can now use flashlights

Gandalf2k15 commented:
Most of these are susceptible to EMP's and will jam & eject mag if hit by an EMP.

Gandalf2k15 commented:
Some rifles now accept bayonets.

Gandalf2k15 commented:
You can clean guns with cloth now! Uses 5 bits to clean!

Kroto-Ker commented:
With all of the new changes and answers to my issues, I believe the new Security will be an excellent edition to the server! I can't commend you enough for how much work you've put into this, and I'm hoping this PR will get through without issue!

hugmeorelse commented:
> As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.
> 
> The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.

I'm already talking to Gandalf on some basic uniform changes like the color scheme. We may also change the overall look of security over time to give them more of a cyberpunk or sci-fi feel to them.

Also, for anyone worried about antagonists, their loadouts and what they get will be reworked over time. We have much planned to enhance the experience, and we're always keeping an open ear to criticism and critique, given that you're being polite and not an asshole about it.

It's amazing how many people initially overreacted to the v1 and v2 changes here before realizing this isn't all that bad.

hugmeorelse commented:
> As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.
> 
> The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.

I'm already talking to Gandalf on some basic uniform changes like the color scheme. We may also change the overall look of security over time to give them more of a cyberpunk or sci-fi feel to them.

Also, for everyone else, I've asked for a security rework for a while now, and Gandalf has delivered on it by doing an excellent job without me having to micromanage, save for the uniform. He has my full backing. It's not a matter of stopping this. It's a matter of addressing any concerns that people have and making tweaks and changes, so that we have excellent content.

Also, for anyone worried about antagonists, their loadouts and what they get will be reworked over time. We have much planned to enhance the experience, and we're always keeping an open ear to criticism and critique, given that you're being polite and not an asshole about it. Your feedback is absolutely crucial in making each big changes we make a success. We will never ignore players, but we will also not cave in at the first sound of dissent. Our focus is making things better with every iteration, until it's ready for the server.

It's amazing how many people initially overreacted to the v1 and v2 changes here before realizing this isn't all that bad. I hope people learned that any planned big approaches by us will have an iterative approach that takes in feedback from players to improve upon the current product.

hugmeorelse commented:
> As I said before, the guns all look very nice, but I'm not sold on the new uniforms/the white and black color scheme in general. I'd be a little mad about removing useful batons from Security, but it gives Cargo a single useful thing to buy, and that can't be all bad.
> 
> The direction this is taking is a little odd to me? It seems like you want Security to be a lot more harsh and potentially lethal. Time will tell how brutal the injuries sustained from rubber bullets are, but this all seems like a big shift in how Security is perceived. They're definitely paramilitary now, not just private security.

I'm already talking to Gandalf on some basic uniform changes like the color scheme. We may also change the overall look of security over time to give them more of a cyberpunk or sci-fi feel to them.

Also, for everyone else, I've asked for a security rework for a while now, and Gandalf has delivered on it by doing an excellent job without me having to micromanage, save for the uniform. He has my full backing. It's not a matter of stopping this. It's a matter of addressing any concerns that people have and making tweaks and changes, so that we have excellent content.

Also, for anyone worried about antagonists, their loadouts and what they get will be reworked over time. We have much planned to enhance the experience, and we're always keeping an open ear to criticism and critique, given that you're being polite and not an asshole about it. Your feedback is absolutely crucial in making each big changes we make a success. We will never ignore players, but we will also not cave in at the first sound of dissent. Our focus is making things better with every iteration, until it's ready for the server. We will ditch an idea if it's not up to scratch. Don't worry about that. We do not have that low of a standard!

On a side note, I'm not surprised on how many people initially overreacted to v1 and v2. I hope people learned that any planned big approaches by us will have an iterative approach that takes in feedback from players to improve upon the current product.

Mithrun commented:
So you can just start with an SMG now? That 's crazy.. overall i think the selection of what you can get get should be narrowed down,but... otherwise really cool!

Gandalf2k15 commented:
> So you can just start with an SMG now? That 's crazy.. overall i think the selection of what you can get get should be narrowed down,but... otherwise really cool!

Its not an smg, its a semi auto pistol rifle.

KasparoVy commented:
Man this is goddamn amazing @Gandalf2k15 . That PR desc is brilliant.

hugmeorelse commented:
New color scheme to be unveiled soon.

Vynellin commented:
Oh god are we going with another color shift?
And id reccomend making the combat medic a bit more prominent maybe give them a medical coat instead with the colors of the new sec and armored.

Gandalf2k15 commented:
> Oh god are we going with another color shift?
> And id reccomend making the combat medic a bit more prominent maybe give them a medical coat instead with the colors of the new sec and armored.

After much debate this is our final choice!

Gandalf2k15 commented:
Update: NEW UNIFORM.

ArmaTek things too.

Couple tweaks.

Gandalf2k15 commented:
Renamed to ArmaDyne.

Oscari-pog commented:
Several potential problems I can see with this
>Are energy-based weapons completely gone or only gone for sec? 
>What about space-loot guns? They are often placed in cohesion to a ruin to help you clear it.
>Are energy based weapons still available in cargo?

H0gberg commented:
> Several potential problems I can see with this
> Are energy-based weapons completely gone or only gone for sec?
> What about space-loot guns? They are often placed in cohesion to a ruin to help you clear it.
> Are energy based weapons still available in cargo?

Energy weapons will still exist, just gone in the armory, space guns will still be fine and energy based I'm pt sure will still exist ya cutescari!

Gandalf2k15 commented:
Added SECURITY SERGEANT:
As part of the security rework and policy change, we've added the security sergeant. A field supervisor with the soul purpose of coordinating security when the HoS is in their office doing nothing. We generally want 1 sergeant for every 3 officers, where possible. This will obviously change depending on how things go.

We really do expect a higher level of roleplay from everyone involved here, and this should make it clear that it's not acceptable to breach policy.

The security sergeant is directly above all security officers and are securities first line of call for any quick decisions. The warden, is for the prison and brig. You now report to the Sergeant for any issues you have.

Sandmoney commented:
Are there plans to add a jumpskirt variant the the standard issue loadout?

Mithrun commented:
The third blue department?

Mithrun commented:
Also i feel like keeping in mind and remembering all the options and their namesa is gonna be hard, especially since the differences are so minute... especially the PDH peacekeeper and the glock where the peacekeeper is just like an upgrade for it?

BrotherHangyul commented:
I kinda liked ArmaTek, gonna be honest. Sounded more snazzy.

still really good PR

YakumoChen commented:
Is this the new Lopland Security?

Really digging these changes overall. Vast improvement over the previous PRs. A corp I can stand behind too, since I played my HoS main as aligned to the previous Security Contractor.
Does the HoS still get the X-01?

Gandalf2k15 commented:
> Is this the new Lopland Security?
> 
> Really digging these changes overall. Vast improvement over the previous PRs. A corp I can stand behind too, since I played my HoS main as aligned to the previous Security Contractor.
> Does the HoS still get the X-01?

Yeup, they still get that for now

hugmeorelse commented:
> Is this the new Lopland Security?
> 
> Really digging these changes overall. Vast improvement over the previous PRs. A corp I can stand behind too, since I played my HoS main as aligned to the previous Security Contractor.
> Does the HoS still get the X-01?

ArmaDyne is the company selling the weapons and uniform. Lopland is still the actual security company that provides the actual personnel.

Gandalf2k15 commented:
> > Is this the new Lopland Security?
> > Really digging these changes overall. Vast improvement over the previous PRs. A corp I can stand behind too, since I played my HoS main as aligned to the previous Security Contractor.
> > Does the HoS still get the X-01?
> 
> ArmaDyne is the company selling the weapons and uniform. Lopland is still the actual security company that provides the actual personnel.

What he said

NonoodWhatever commented:
The Retro laser gun type will still be there?

unit0016 commented:
A few things:
First, AI laws.
As an ex-AI main. The laws should be special case "oh only this lawset is equal". What about Ion laws. Hacked laws? How am I supposed to the AI in a clever way in a lawset that would conflict with most anything I write, because they're all equal? How is the AI supposed to act when given a conflicting law? Is it suddenly not equal then? Would that not throw stealthy AI subversions out the window?

Point is, no, it shouldn't be equal. A better solution would be to rewrite law one to specifically define asset damage, so that people just building a sauna don't get locked in and security called.

Second, security's color scheme.
Something about it just.. garish. It feels you threw a blue tint over the red sprites without making it monochrome first, creating this weird murky almost green but not quite blue? 
Doesn't look good. Same with the new mindshield outline. The green looks just as garish, especially paired with the weird color security already has. Why not.. white? Or, alternatively, since all the job icons are white - why not change how it's signified entirely? Maybe a white dot under the normal sechud icon like what nanites have on med and sechuds.

Gandalf2k15 commented:
## Bullet drive
You know those pesky bullet casings? Fuck em. They're annoying. So we at Armadyne present the BulletDriveTM, it'll suck away all your issues.
![image](https://i.imgur.com/bBwYQud.png)

Warning: Misuse may lead to accidental limb removal.

Gandalf2k15 commented:
> A few things:
> First, AI laws.
> As an ex-AI main. The laws should be special case "oh only this lawset is equal". What about Ion laws. Hacked laws? How am I supposed to the AI in a clever way in a lawset that would conflict with most anything I write, because they're all equal? How is the AI supposed to act when given a conflicting law? Is it suddenly not equal then? Would that not throw stealthy AI subversions out the window?
> 
> Point is, no, it shouldn't be equal. A better solution would be to rewrite law one to specifically define asset damage, so that people just building a sauna don't get locked in and security called.
> 
> Second, security's color scheme.
> Something about it just.. garish. It feels you threw a blue tint over the red sprites without making it monochrome first, creating this weird murky almost green but not quite blue?
> Doesn't look good. Same with the new mindshield outline. The green looks just as garish, especially paired with the weird color security already has. Why not.. white? Or, alternatively, since all the job icons are white - why not change how it's signified entirely? Maybe a white dot under the normal sechud icon like what nanites have on med and sechuds.

Uniform sprites are by JungleRat, the tactical one and stuff.

Gandalf2k15 commented:
You can now buy weapon cleaning kits from the vendor
Rubber and normal bullets are now standard in the protolathe
Misc fixes

Gandalf2k15 commented:
All guns now come with 4mags.

GoldenAlpharex commented:
I just want to ask, why did the mindshield HUD icon need to be turned green all of a sudden? It was really good-looking before and now it just looks... Off. There's nothing green about security, it just feels like a "we need to change the color of this" kind of deal, whilst I really don't think it was necessary. Otherwise, so far, it seems to be pretty interesting? I'll probably have to give Security a try at some point to see how it is, but that's just what I've noticed so far.
Oh, and, good job on the sprites, and thank you for not leaving them white, they look a lot more ominous now, I like it.

Gandalf2k15 commented:
> I just want to ask, why did the mindshield HUD icon need to be turned green all of a sudden? It was really good-looking before and now it just looks... Off. There's nothing green about security, it just feels like a "we need to change the color of this" kind of deal, whilst I really don't think it was necessary. Otherwise, so far, it seems to be pretty interesting? I'll probably have to give Security a try at some point to see how it is, but that's just what I've noticed so far.
> Oh, and, good job on the sprites, and thank you for not leaving them white, they look a lot more ominous now, I like it.

Reverted

GoldenAlpharex commented:
> > I just want to ask, why did the mindshield HUD icon need to be turned green all of a sudden? It was really good-looking before and now it just looks... Off. There's nothing green about security, it just feels like a "we need to change the color of this" kind of deal, whilst I really don't think it was necessary. Otherwise, so far, it seems to be pretty interesting? I'll probably have to give Security a try at some point to see how it is, but that's just what I've noticed so far.
> > Oh, and, good job on the sprites, and thank you for not leaving them white, they look a lot more ominous now, I like it.
> 
> Reverted

Oh, really? Thanks, that's sick. :)

SarmentiCampbell commented:
So, I just got one totally unplayable issue with it and just a question for it,
The issue is that when you hold the peacekeeper duffelbag in your hand, it turn into the red bag sprite one, literally unplayable. /s

The second one, Uh Would Detective get a rework aswell along with new clothes and all?

Gandalf2k15 commented:
> So, I just got one totally unplayable issue with it and just a question for it,
> The issue is that when you hold the peacekeeper duffelbag in your hand, it turn into the red bag sprite one, literally unplayable. /s
> 
> The second one, Uh Would Detective get a rework aswell along with new clothes and all?

That's coming next, the detpart.

SarmentiCampbell commented:
hell yeah

YakumoChen commented:
My only concern looking at the sprites is that the Warden is naught but a few hues away from a secoff. A more substantial color difference I think is needed, and/or they should default to the sergeant cap to make it clear their rank is the same.
Ideally they should be wearing more red, obstinately the "brig color", and the medic should be wearing a hue closer to what medbay has to make their medic status more obvious. Something like a lighter blue or green.

unit0016 commented:
> 
> Uniform sprites are by JungleRat, the tactical one and stuff.

mainly talking about the doors and mindshield, I mean.

unit0016 commented:
![image](https://user-images.githubusercontent.com/50649185/106342050-6aeff480-626d-11eb-99d2-03b425ec1e53.png)
Noticed this error while codediving, henk

KathrinBailey commented:
![image](https://user-images.githubusercontent.com/53862927/106344847-2764ad00-62a4-11eb-946f-5fda23e401da.png)
Could do with a separate icon_state for empty boxes, this gets really confusing.

YakumoChen commented:
security's HUD icons look close to unreadable for some reason, I can't tell if it's the mindshields or if the icons are just changed to look like absolute ass.

Gandalf2k15 commented:
Moved protolathe to armory, added bulletbot! Added rubbers and IHDF toa  vending machine.

![image](https://i.imgur.com/AV3qD0Z.png)

YakumoChen commented:
The other thing I noticed was that the brig paint scheme makes it _very_ dark. It is quite jarring to go towards the brig and cross the paint line, it feels like you're thrust into the Underdark and the dark tiles feel unnecessary and overpowering.

Gandalf2k15 commented:
Primary armament holochips  no longer redeemable on < red alert

GoldenAlpharex commented:
Say, would it be possible to have a more visible color for the Security comms? It's super hard to read when it's blue like that, alongside all of the <span='notice'> text (I think that's the blue one). Maybe even just as a preference for the Security color, that'd be pretty sick I think.

YakumoChen commented:
I'd like to see a return to more red colors, like on the floor and on radio. The Blue-ening is pretty excessive, and it's especially bad when Security Blue is the same color as Command radio on a lot of code bases.

Gandalf2k15 commented:
Added the security hardsuit, courtecy of JungleRatings!

NotRanged commented:
Seriously consider making everything (and balancing it around) hitscan. At high tidi, you can outrun and even run into your own bullets.

YakumoChen commented:
The old head of security jumpsuits are broken on wearing.

YakumoChen commented:
ED209 is unbuildable due to lack of disablers obtainable.

YakumoChen commented:
![image](https://user-images.githubusercontent.com/10399117/106691092-d1a54300-65ca-11eb-9997-4dad6e9e788c.png)

YakumoChen commented:
The "missing person" poster in the security office of one of the maps is a broken item. The gun cases also appear to have a debug description.

Gandalf2k15 commented:
Added a new hitscan lasergun instead of the old laser gun, they're found in the armory.

Gandalf2k15 commented:
Full merging for future proofing.

Kazkin commented:
These sprites look really familiar...

Enzoman12 commented:
What happened with the revolvers? Minor thing but I liked them alot. Did they get permanently removed or something?

LeonY24 commented:
damn this rework is hell of a thing, BUT
armadyne seems to be supplying lopland with civil-gunpowder ammo, cuz guns are needed to be cleaned (which is kinda long and unhandy on emergencies) too much for their effeciency, which's kinda low - including dodging, firefights and shit i have to spend ~1,5 mags on Glock-17.

YakumoChen commented:
Budget cuts
